### DISCUSSION TITLE
Index selector does not work for IEnumerables

### DISCUSSION THREAD

**Discussion by dioguima** [axuno/SmartFormat#312] [Q&A]
Hello,

I am updating the version of SmartFormat nugget package to version 3.1 (before it was 1.6.1) and I'm having problems making the "Index" selector work. Does anyone know if it was replaced in the newer versions? I was not able to find the documentation for it

**Comment by axunonb** [comment]
You mean how to access a list element?
This is [mentioned in the wiki](https://github.com/axuno/SmartFormat/wiki/Syntax%2C-Terminology#output-a-list-element)
Or is it [how to synchronize 2 lists](https://github.com/axuno/SmartFormat/wiki/Lists-_-ListFormatter#synchronize-2-lists)?

Otherwise please give code sample.

**Reply by dioguima** [reply]
It is similar to the second option, actually, I just realized what could be the problem, it seems that the way that the collection is passed to the format method, I was able to isolate the problem with the code sample below, the second format will throw the error

```
using SmartFormat;

var objects = new object[]
{
    new { },
    new { },
};
Console.WriteLine(Smart.Format("{1:S{Index}|;}", null, objects));
Console.WriteLine(Smart.Format("{1:S{Index}|;}", null, objects.Select(x => x)));
```

**Reply by axunonb** [reply]
In the second case, the object is an `Enumerable`, which cannot be handled by the `ListFormatter`.
In `v1.6.1 ` a `Selector` was tested for having the name "index", even if data was not an `IList`.
This is currently not the case in `v3.1.0` and will be brought back.

**Reply by dioguima** [reply]
Thank you
